src/neo/SmartContract/ApplicationEngine.Storage.cs

```diff
@@ -165,8 +165,12 @@ protected internal IIterator Find(StorageContext context, byte[] prefix, FindOpt
         /// <param name="value">The value of the entry.</param>
         protected internal void Put(StorageContext context, byte[] key, byte[] value)
         {
-            if (key.Length > MaxStorageKeySize || value.Length > MaxStorageValueSize || context.IsReadOnly)
-                throw new ArgumentException();
+            if (context.IsReadOnly)
```

Comment:
I don't think ANY exception in methods that modify contract storage should be catchable